### DISCUSSION TITLE
Unable to use given session dependency from example code

### DISCUSSION THREAD

**Discussion by GilbertBckr** [fastapi-users/fastapi-users#1018] [Q&A] [answered]
After copying the full example for sqlalchemy from the documentation I wanted to add my own models to the database. But somehow the session dependency isnt working. Everytime I try to use it get an Attribute Error: AttributeError: 'Depends' object has no attribute 'add'. I would be very grateful if anyone could help me with this issue.

This is my db.py code:

`
from typing import AsyncGenerator

from fastapi import Depends
from fastapi_users.db import SQLAlchemyBaseUserTableUUID, SQLAlchemyUserDatabase
from sqlalchemy import Column, Integer, String
from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
from sqlalchemy.ext.declarative import DeclarativeMeta, declarative_base
from sqlalchemy.orm import sessionmaker

DATABASE_URL = "sqlite+aiosqlite:///./test.db"
Base: DeclarativeMeta = declarative_base()

class User(SQLAlchemyBaseUserTableUUID, Base):
    pass

class Test(Base):
    __tablename__ = "test"
    id = Column(Integer, primary_key=True)
    name = Column(String())

engine = create_async_engine(DATABASE_URL)
async_session_maker = sessionmaker(engine, class_=AsyncSession, expire_on_commit=False)

async def create_db_and_tables():
    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)

async def get_async_session() -> AsyncGenerator[AsyncSession, None]:
    async with async_session_maker() as session:
        yield session

async def get_user_db(session: AsyncSession = Depends(get_async_session)):
    yield SQLAlchemyUserDatabase(session, User)

async def create_test(session: AsyncSession = Depends(get_async_session)):
    test = Test(name="test")
    session.add(test)
    session.commit()`

**Comment by frankie567** [comment, marked as answer]
Your function `create_test` won't work because `Depends` is a special object to perform dependency injection *inside* FastAPI. Only FastAPI knows how to handle this inside a router.

If you want to call those functions in a script or whatever outside FastAPI, you'll have to provide the dependencies values yourself. There is some rationale about this here: https://fastapi-users.github.io/fastapi-users/10.0/cookbook/create-user-programmatically/

**Reply by GilbertBckr** [reply]
Ah my bad. Thanks a lot for your help.
